contracts/utils/contracts/src/Refundable.sol

```diff
@@ -19,6 +19,18 @@
 pragma solidity ^0.5.9;
 
 
-interface IStaking {
-
+contract Refundable {
+
+    // This bool is used by the refund modifier to allow for lazily evaluated refunds.
+    bool internal shouldNotRefund;
+
+    modifier refund {
+        if (shouldNotRefund) {
+            _;
+        } else {
+            shouldNotRefund = true;
```

Comment:
It looks like calling this modifier twice will actually disable refunds if I'm reading this correctly? How would they be turned back on?

We can also implement this without requiring an SSTORE for the default case. All this really needs to do is:

```
_;
if (!shouldNotRefund) {
    msg.sender.transfer(address(this).balance);
}
```

This only requires adding the cost of an SLOAD to every single function that uses the modifier, with the exception of the batch `noThrow` functions (only 3 functions I believe). For those we will have to set `shouldNotRefund` to `true` at the beginning of the function call and reset it to `false` at the end of the function call. That can actually probably be its own modifier in this contract (maybe call it `disableRefundsUntilEnd` or something similar).